src/utilities/graphqlConnectionFactory.ts

```diff
@@ -36,3 +38,154 @@ export function graphqlConnectionFactory<T>(): InterfaceConnection<T> {
     },
   };
 }
+
+// Type definition for a mapping funtion
+type GetNodeFromResultFnType<T, U> = {
+  (result: U): T;
+};
+
+// A custom type for easier implementation of the business logic of the connection factory
+export type ConnectionArguments = {
+  cursor: string | null | undefined;
+  direction: "BACKWARD" | "FORWARD";
+  limit: number;
+};
+
+// Utility to convert the recieved arguments to the ConnetionArguments type
+export const transformArguments = (
+  args: CursorPaginationArgsType
+): ConnectionArguments => {
+  const transformedArgs: ConnectionArguments = {
+    cursor: undefined,
+    direction: args.first ? "FORWARD" : "BACKWARD",
+    limit: args.first | args.last,
+  };
+
+  if (args.after) transformedArgs.cursor = args.after;
+  else if (args.before) transformedArgs.cursor = args.before;
+
+  return transformedArgs;
+};
+
+// Generates the limit that can should be passed in the .limit() method
+export const getLimit = (args: ConnectionArguments) => {
+  // We always fetch 1 object more than  args.limit
+  // so that we can use that to get the information about hasNextPage / hasPreviousPage
+  // When args.cursor is supplied, we fetch 1 more object so as validate the cursor as well
+  return args.cursor ? args.limit + 2 : args.limit + 1;
+};
+
+// Generates the sortingObject that can be passed in the .sort() method
+export const getSortingObject = (
+  args: ConnectionArguments,
+  sortingObject: { [key: string]: number }
+) => {
+  // We assume that the resolver would always be written with respect to the sorting that needs to be implemented for forward pagination
+  if (args.direction === "FORWARD") return sortingObject;
+
+  // If we are paginating backwards, then we must reverse the order of all fields that are being sorted by.
+  for (const [key, value] of Object.entries(sortingObject)) {
+    sortingObject[key] = value * -1;
+  }
+
+  return sortingObject;
+};
+
+// Generates the sorting arguments for filterQuery that can be passed into the .find() method
+export function getFilterQuery(args: ConnectionArguments) {
```

Comment:
Maybe name this one `getFilterObject`

Reply:
@xoldyckk 
I get your concerns, but I don't think it would be feasible to create utilities that can handle the complex nature of queries that have been highlighted in the later section of the attached link. It would make sense then that the developer himself writes the complete sorting and filtering logic himself.

Also currently, the filtering and sorting logic is strictly opt in only, as the `getSortingObject` function only return the `$lte` or `$gte` thingie if the `cursor` in supplied and the common practice of sorting by `_id` is used. If the same is not to be used, then these utilities can be ignored and a manual implementation can be done.

Reply:
@xoldyckk Can you please review?

Reply:
@EshaanAgg I thought we were waiting for standardization of errors. Without that I can't review as it'd be wasted effort.